src/messages/Image.cpp

```diff
@@ -476,4 +517,97 @@ bool Image::operator!=(const Image &other) const
     return !this->operator==(other);
 }
 
+ImageExpirationPool::ImageExpirationPool()
+{
+    QObject::connect(&this->freeTimer_, &QTimer::timeout, [this] {
+        if (isGuiThread())
+        {
+            this->freeOld();
+        }
+        else
+        {
+            postToThread([this] {
+                this->freeOld();
+            });
+        }
+    });
+
+    this->freeTimer_.start(
+        std::chrono::duration_cast<std::chrono::milliseconds>(
+            IMAGE_POOL_CLEANUP_INTERVAL));
+}
+
+ImageExpirationPool &ImageExpirationPool::instance()
+{
+    static ImageExpirationPool instance;
+    return instance;
+}
+
+void ImageExpirationPool::addImagePtr(ImagePtr imgPtr)
+{
+    std::lock_guard<std::mutex> lock(this->mutex_);
+    this->allImages_.insert(
```

Comment:
Why not use `this->allImages_.emplace(
        imgPtr.get(), std::weak_ptr<Image>(imgPtr));`?